File: RedSheet.DbEntities/Models/vCommunicationMode.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vCommunicationModes", Schema = "dbo")]
	[Serializable]
	public partial class vCommunicationMode  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CommunicationModeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string CommunicationModeName { get; set; } 

	public vCommunicationMode()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-communication-mode.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vCommunicationMode {
    constructor(vCommunicationMode?: vCommunicationMode )  {
        let properties = [ "className", "communicationModeId", "communicationModeName",];
        for (let property of properties)
            if (vCommunicationMode && vCommunicationMode[property])
                this[property] = vCommunicationMode[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	communicationModeId : number =   0 ;
 
    @required()
    @maxLength(200)
	communicationModeName : string =   undefined;


}
